packages/backend/migrations/20210715110049_create_payment_progress_table.js

```diff
@@ -0,0 +1,15 @@
+exports.up = function (knex) {
+  return knex.schema.createTable('paymentProgress', function (table) {
+    table.uuid('id').notNullable().primary() // This is identical to OutgoingPayment.id
+
+    table.bigInteger('amountSent').notNullable()
+    table.bigInteger('amountDelivered').notNullable()
```

Comment:
Originally, the reason was so that the "progress" could be updated whenever a stream packet was fulfilled, while the payment row "lock" was still held. (The "lock" is a postgres transaction with `SELECT ... FOR UPDATE`). That way, even if the transaction failed to commit, the payment progress would not be lost.

This approach had some issues, though:

- Handling of failed `PaymentProgress` updates could lead to double-send or failed payments.
- The `PaymentProgress`'s `amountSent` could get out of sync with the true amount sent, tracked by Tigerbeetle.
- The `amountDelivered` could get out of sync with the true amount delivered, tracked by the remote invoice (for FixedDelivery aka invoice payments, that is). For FixedSend payments, `amountDelivered` is fuzzy anyway.

So instead, I'm changing things around (not committed yet):

  - Remove the payment progress table.
  - Remove the `OutgoingPayment.outcome.amountDelivered` field. For `FixedDelivery` payments, the client can just fetch it from the invoice.
  - The `OutgoingPayment.outcome.amountSent` will come directly from TigerBeetle, so they can't ever get out of sync.